src/software/thunderscope/robot_diagnostics/error_log_widgets.py

```diff
@@ -0,0 +1,273 @@
+from pyqtgraph.Qt.QtWidgets import *
+from pyqtgraph.Qt.QtCore import Qt
+from pyqtgraph.Qt import QtGui
+from proto.import_all_protos import *
+from datetime import datetime
+import textwrap
+import software.thunderscope.robot_diagnostics.robot_error_log_icons.error_log_constants as error_constants
+from proto.robot_log_msg_pb2 import RobotLog, LogLevel
+from software.thunderscope.robot_diagnostics.robot_status import RobotStatusView
+from software.thunderscope.constants import LOG_LEVEL_STR_MAP
+
+
+class RobotLogMessageWidget(QFrame):
+    """
+    A single error message from a robot.
+
+    Displays an icon corresponding to the type of error, the robot id, the message
+    and the timestamp of the error
+    """
+
+    def __init__(
+        self, robot_id: int, message: str, icon: QtGui.QPixmap, timestamp=None
```

Comment:
I think we should consider making this error widget generalizable enough to also work for error logs from fullsystem and simulator processes where there is not robot id. Possibly instead of a `robot_id` we can have a `title: str` field. The class name can then be simplified to something like `ErrorLogMessageWidget`. 

Ideally we would also update the `RobotLog` protobuf message to be more general `Log` message, since we also use it in fullsystem and simulator. Though let me know if you think these are better suited for a different PR:
- [ ] `RobotLog` protobuf message type should be updated to remove `robot_id` (safely deprecated using `reserved = 1` `reserved "robot_id"`) and replace it with a more general `string source = 7` field. This field can then be filled in `NetworkLoggerSingleton` -> `NetworkSink` to have a value of `R<robot_id>`, similarly `LoggerSingleton` -> `ProtobufSink` can be updated to take in `Blue`/`Yellow`/`Sim` as the source. The protobuf message type could also be renamed to have a more generic name like `Log`.
- [ ] `g3log_widget.py` print the `log.source` instead of `log.robot_id`
- [ ] Add `RobotLog` to `Simulator.setup_proto_unix_io` so logs from the simulator are also received by Thunderscope.

Since this would be a relatively major change of an existing feature, @itsarune what are your thoughts?

Reply:
that's a good idea, I think that'll be an improvement. If @sauravbanna doesn't have too much on his plate, then we can include it in this PR